### PR TITLE
chore: prepare AISimulate 0.13.0 release

### PR DESCRIPTION
**Pull request by saturley-hall**
## Why and what changed

Prepare AISimulate 0.13.0 by synchronizing wheel/crate versions, lockfiles, release checks, and artifact documentation. Retain the `aiconfigurator` compatibility command in 0.13.0 and target removal for 0.14.0 once all remaining workflows have verified unified-CLI replacements. Align current examples and runtime warnings; preserve dated publication records.

Repair release validation exposed by integrating current main. Keep workflow-only contracts in Fast CI and installed-wheel FPM accuracy coverage in Full CI, with joint application/FPM dependency resolution and `pip check`. Fetch the numerical manifest's immutable historical commit when a fresh clone lacks it after squash merge. Align Rust fixtures, the backend-facts registry, the wheel verifier, and replay tests with merged #244 and #242.

## Review map

- Risk level: medium, because this release bump also repairs CI execution contracts and stale test/reference fixtures.
- Start with: `scripts/check_prediction_numerics.py` and the two active CI workflows; `aiconfigurator/deprecation.py` and migration docs; the Rust fixture loaders, wheel SDK verifier, and canonical replay assertions. The Collector registry diff is limited to 221 additions and six removals traced to #244.
- Public or serialized contract changed: artifact version becomes 0.13.0 and compatibility-command messaging names the 0.14.0 removal target. No production schema, estimator math, performance measurements, or import surface is changed by this PR.
- Compatibility or rollback concern: retain the established command and arguments through 0.13.0; removal remains conditional on verified replacements. Imports remain available. Numerical baseline SHA, expected values, tolerances, and engine goldens are unchanged. This PR does not publish a release or claim downstream Dynamo qualification.

## Evidence

- Tests: `cargo test -p aisimulate-core --lib perfmodel::` passed 698 tests (one existing ignored test). `python -m pytest -p no:timeout tests/test_afd_artifacts.py tests/test_unified_traffic_runtime.py tests/e2e/test_unified_cli_engine.py -q` passed 65 tests. Backend-facts, package-layering, and deprecation unit suites passed 45 tests. `python -m pytest -p no:timeout tests/test_ci_qualification.py -q` passed all 40 tests on the final tree.
- Additional local checks: locked/offline Cargo metadata and uv lock checks; `scripts/build_release_artifacts.py --check-only`; Ruff check/format; `cargo fmt --all --check`; `git diff --check`; current SDK-facade/native-engine smoke in the editable environment. Earlier workflow/FPM validation passed 356 cases (three installed-native cases deferred to hosted CI), and the earlier Fast CI test set passed 304 cases.
- Fast CI: [35361857149](https://github.com/ai-dynamo/aisimulate/actions/runs/35361857149) passed on `031d8db3a81e4e42ffe43c37161517f2ec05157d`.
- Full CI: [35363166376](https://github.com/ai-dynamo/aisimulate/actions/runs/35363166376) passed all 54 jobs on `031d8db3a81e4e42ffe43c37161517f2ec05157d`, including both architectures, all three platform wheels, installed-wheel FPM/replay contracts, Collector validation, and engine/prediction regression gates.
- CodeRabbit reviewed commit: `031d8db3a81e4e42ffe43c37161517f2ec05157d`, completed with no actionable code comments. Its description-template warning is addressed by these sections.
- Codex reviewed commit: `031d8db3a81e4e42ffe43c37161517f2ec05157d`, no remaining actionable findings after the full diff and follow-up review.
- Negative or boundary cases: real shallow-Git baseline fetch preserves HEAD/branch, works offline once cached, rejects invalid SHAs, and fails for unavailable commits. Direct CLI tests require exactly one mode and verify baseline-only execution needs no prediction cases and writes no results. Warning deduplication and legacy-call forwarding remain covered.
- Expected-value derivation: no predicted numerical values are updated. The AFD serialization digest changes from `bbbd762b...` to `3f53f3ae...`; removing only #242's newly serialized empty `forward_pass_estimators` field exactly reproduces the old digest.
- Before/after output: Rust fixture assertions now execute through the declared reuse path; replay tests assert per-role `timing_model.config`; wheel verification expects the two canonical public FPM exports. All other AFD golden content is identical.

## Modeling or data provenance

No new measurements, proxies, or production modeling changes. All 227 changed Collector facts map to parquet files changed by #244: 221 vLLM 0.25.0 slices (49 each B300/GB200/GB300; 37 each H100/H200) and six deliberately removed direct 0.24.0 FP8-block GEMM slices. Existing backend/kernel mappings and the curated translation map are unchanged. `python tools/perf_database/backend_facts.py --check` passes for all 2,137 slices. The collection provenance, physical-key and anomaly checks, and declared-reuse limits remain documented in [the #244 evidence bundle](docs/data/pr244/README.md).

## Tracking

- Closes: N/A; release preparation.
- Related PRs or issues: [#244](https://github.com/ai-dynamo/aisimulate/pull/244), [#242](https://github.com/ai-dynamo/aisimulate/pull/242), and the compatibility work tracked in the migration guide.

### CODE REVIEW

File: python/aisimulate/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ SPDX-License-Identifier: Apache-2.0
 # AISimulate application (AIConfigurator compatibility source)
 
 > This directory contains the complete AIConfigurator application migrated to
-> the standalone AISimulate repository. It builds the `aisimulate` 0.12.0
+> the standalone AISimulate repository. It builds the `aisimulate` 0.13.0
```
Comment (jasonqinzhou) [inline python/aisimulate/README.md:9]: The 0.13.0 wheel retains the compatibility command and import namespaces, but README.md and the migration/history documentation still target command removal in 0.13.0. Jason has confirmed the command should remain through 0.13.0 with removal targeted for 0.14.0. Align the current compatibility documentation and active command warning with that policy, without changing dated 0.12.0 release snapshots.

File: docs/artifact-contract.md [hunk 2]
```diff
@@ -36,8 +36,8 @@ fails CI if either packaging copy differs byte-for-byte from its root original,
 and the release-artifact validator checks the bytes installed in the wheel.
 
 Nightly builds stamp a dev suffix with `scripts/apply_dev_version.py` before
-building: the wheel becomes `0.12.0.devYYYYMMDD` (PEP 440) and the crate
-`0.12.0-dev.YYYYMMDD` (SemVer — cargo rejects the PEP 440 spelling, and the
+building: the wheel becomes `0.13.0.devYYYYMMDD` (PEP 440) and the crate
```
Comment (jasonqinzhou) [inline docs/artifact-contract.md:39]: The artifact contract now uses the 0.13.0 nightly line, but current nightly guidance in docs/ci.md, docs/installation.md, and the apply_dev_version.py docstring still uses 0.12.0. Update these current examples alongside the version bump while preserving historical fixtures and publication records.

**Review by jasonqinzhou — COMMENTED** [review]
The version metadata and lockfiles are synchronized, and the five focused nightly-version/historical-manifest tests pass locally with a working Node runtime. Before closeout, the contracts jobs need their missing tools installed: Full CI run 35304255738 fails on both architectures with missing huggingface_hub during FPM test collection and missing Node.js in _nightly_version. These requirements predate this PR. I will add explicit contract-job setup and rerun the exact updated head.

File: python/aisimulate/README.md (hunk 1 above)
Comment (jasonqinzhou) [reply python/aisimulate/README.md:9]: Fixed in 313af7c3d5ec4fad6d6ce36251d4d4bb8e97c16b: the command remains in 0.13.0, removal is targeted for 0.14.0 subject to verified replacements, and current docs/runtime warnings agree. The warning now points to the maintained AISimulate migration guide; current import compatibility is documented without promising a new import-removal date.

File: docs/artifact-contract.md (hunk 2 above)
Comment (jasonqinzhou) [reply docs/artifact-contract.md:39]: Fixed in 313af7c3d5ec4fad6d6ce36251d4d4bb8e97c16b: current nightly examples use the 0.13.0 line in the CI/install guides and stamping script. Dated 0.12.0 publication records and historical fixtures remain unchanged.

**Review by jasonqinzhou — COMMENTED** [review]
Current-main integration exposed a separate reproducible CI provenance setup failure: Fast CI 35357873986 cannot resolve numerical baseline af91885dfdeddedebd256c0c4df11a6f3e75e97e. GitHub still serves this historical PR commit, but normal branch checkout does not include it after squash merge. Preserve the manifest SHA, all eight expected values, and their tolerances; explicitly fetch that validated immutable baseline before Fast CI contract tests, Full CI repository contracts, and native numerical qualification.

**Comment by jasonqinzhou** [comment]
@coderabbitai full review

**Review by jasonqinzhou — COMMENTED** [review]
Codex review of 81e214e5121bb4a902b1efb7154eb809ce5856a6 against current main d9f1580aa4385e6f2199e1f189b64cf7528009d5: no remaining actionable findings. Product and artifact versions agree at 0.13.0; the legacy command remains available with removal targeted for 0.14.0 subject to verified replacements. Current docs and warnings agree, and historical publication records remain intact. The numerical-baseline repair validates the full SHA, explicitly fetches only missing commits, preserves checkout HEAD/branch and the frozen manifest, and fails on unavailable sources. Local validation includes 356 workflow/FPM cases (three native-runtime cases deferred to hosted CI), all 304 Fast CI workflow/qualification/release-FPE cases after the fetch fix, release-manifest/lock checks, lint, and warning behavior. Hosted Fast CI 35358464789 passed on this SHA. Current-head CodeRabbit and Full CI remain pending; this is not merge readiness.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -667,14 +663,15 @@ jobs:
...
-          python -m pip check
+          python scripts/check_prediction_numerics.py --fetch-baseline-only
           python scripts/check_application_test_inventory.py --output application-test-inventory.json
           python -m pytest -q \
             python/aisimulate/tests/unit/cli/test_aisimulate_compatibility.py \
             python/aisimulate/tests/cross_package
           aiconfigurator version
-          python -m pytest -q -c pytest.ini tests -n 4
+          python -m pytest -q -c pytest.ini tests -n 4 \
+            --ignore=tests/fpm_accuracy \
```
Comment (jasonqinzhou) [inline .github/workflows/ci.yml]: [P2] Keep installed-wheel FPM tests in Full CI. GitHub Pages runs this suite only for selected paths and does not install AISimulate, so test_real_regression_uses_canonical_identity_and_options and the two canonical adapter cases skip there. Excluding the entire directory here removes their native-runtime PR coverage, including for package/API changes that do not trigger Pages. Retain the FPM tests and their pinned dependency inputs plus pip check in the contracts shard; workflow-only tests can remain delegated to Fast CI.

**Review by jasonqinzhou — COMMENTED** [review]
The workflow-contract exclusion can rely on the existing exact-head Fast CI coverage, but the FPM accuracy exclusion removes installed-wheel coverage and needs correction before merge.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (jasonqinzhou) [reply .github/workflows/ci.yml]: Fixed in dd23ccbff25928561fad3bb9e13fad8ab885412a: Full CI retains installed-wheel FPM tests, joint dependency resolution, and pip check. Workflow-only contracts remain excluded from Full CI because exact-head Fast CI runs them, so the redundant Node.js setup stays removed. The focused workflow coverage test passes.

**Comment by jasonqinzhou** [comment]
@coderabbitai full review

**Review by jasonqinzhou — COMMENTED** [review]
Full CI on dd23ccb exposed two inherited contract mismatches that need correction before approval:

- [P1] Pure-Rust engine/FPM test fixtures load only direct b200_sxm/vllm/0.24.0 tables, but the corrected database now supplies fp8_block through its declared reuse layer. Sixteen tests fail before reaching their behavioral assertions. Load the fixtures through the supported reuse-aware database path; preserve assertions and all production data/math.
- [P1] The installed-wheel verifier still expects the pre-migration SDK facade, rejecting the newly public ForwardPassPerfModelConfig and ForwardPassPerfOptions on all three wheel platforms. Update its exact export contract to the current canonical SDK.

The prediction comparison and engine golden gates already pass. I am investigating the remaining Full CI failures before pushing the corrections.

**Review by jasonqinzhou — COMMENTED** [review]
Further Full CI findings on dd23ccb:

- [P1] Collector backend facts were not synchronized with merged PR #244. All 227 differences map to parquet files changed by ba64235a: 221 new vLLM 0.25.0 slices (B300/GB200/GB300/H100/H200) and six intentionally removed direct 0.24.0 FP8-block GEMM slices. Existing shared slices have no changed backend/kernel mappings. Update only those facts; preserve the curated translation map, measurements, and strict drift check.
- [P1] Three replay contract tests still assert retired aic_forward_model/aic_system rank arguments after PR #242 moved that identity into forward_pass_estimators. Assert canonical per-role estimator identity instead. The AFD artifact golden also needs its replay-spec hash updated for the newly serialized empty forward_pass_estimators field; verify that this field alone explains the digest change before updating it. No performance golden values need to change.

**Review by jasonqinzhou — COMMENTED** [review]
Codex review of bc222e660f0a60e927e039b13854bebe3f1d5a19 against main d9f1580aa4385e6f2199e1f189b64cf7528009d5: no remaining actionable findings in the complete diff. The inherited failures reported above are corrected in this commit. Clarification: prediction rank identity is asserted through canonical timing_model.config; forward_pass_estimators is the additional serialized deployment field responsible for the AFD digest change.

Local evidence: 698 Rust performance-model tests passed (one existing ignored test), 65 replay/CLI tests passed, 45 registry/layering/deprecation tests passed, and the strict registry check passed for 2,137 slices. Every registry change maps to a parquet file changed by #244; existing mappings and production measurements are untouched. Removing only the new empty serialization field exactly reproduces the old AFD hash, with all other golden content equal. SDK export/native-engine smoke and lint/format/whitespace checks pass. Earlier baseline-fetch regression coverage remains in the final tree.

Release versions remain 0.13.0 and the retained compatibility command targets removal in 0.14.0 subject to verified replacements. Numerical sentinel baselines, values, tolerances, and engine goldens remain unchanged. Final-head CodeRabbit and hosted CI are still required before the requested approval and merge.

**Review by jasonqinzhou — COMMENTED** [review]
Validated the two warnings in CodeRabbit's completed dd23ccb review against current bc222e6:

- [P2] The active API mapping and temporary Cargo alias in docs/repository-history.md still target 0.12 even though the current release is 0.13. Update those current-use examples while preserving the explicitly historical migration-start version.
- [P2] The new baseline-fetch helper has real Git coverage, but its CLI mode-selection boundary lacks a direct regression test. Add subprocess coverage proving --fetch-baseline-only succeeds with only a baseline manifest (no prediction cases), produces no prediction artifact, and rejects missing or conflicting mode arguments.

Both are narrow release-validation corrections and will be fixed before final approval.

**Review by jasonqinzhou — COMMENTED** [review]
Codex review refreshed for 031d8db3a81e4e42ffe43c37161517f2ec05157d against unchanged main d9f1580a: no remaining actionable findings. The two CodeRabbit cross-layer warnings are fixed. The latest delta updates current-use 0.13 compatibility examples and adds direct CLI boundary coverage; all 40 numerical-qualification tests, lint, and whitespace checks pass. The full-diff assessment and 698 Rust / 65 replay-CLI / 45 layering-registry-deprecation test evidence from bc222e6 remain applicable to unchanged files. Final-head CodeRabbit, Fast CI, and Full CI are still required before approval and merge.

**Comment by jasonqinzhou** [comment]
@coderabbitai review

Please refresh coverage for the current head `031d8db3a81e4e42ffe43c37161517f2ec05157d`. Both cross-layer warnings are fixed: active API/Cargo examples now target 0.13 while historical migration milestones stay dated, and direct subprocess tests cover `--fetch-baseline-only`, no prediction cases/output, and missing/conflicting mode arguments. All 40 numerical-qualification tests pass. The preceding commit also corrects inherited Full CI fixture, SDK-verifier, replay-contract, and registry mismatches, with validation recorded in the PR description.

**Comment by jasonqinzhou** [comment]
@coderabbitai The description-template warning is addressed. The PR body now includes every required section and field: review map, medium risk, contract/compatibility impact, exact reviewed commit, commands and results, negative cases, digest derivation, data provenance, and tracking. Full CI is running on the unchanged reviewed head 031d8db3a81e4e42ffe43c37161517f2ec05157d: https://github.com/ai-dynamo/aisimulate/actions/runs/35363166376. Please recheck the description warning.

**Review by jasonqinzhou — APPROVED** [review]
Approved after the requested release/compatibility corrections and exact-head verification. The compatibility command remains in 0.13.0 with removal targeted for 0.14.0 after verified replacements. CodeRabbit and Codex reviewed 031d8db3a81e4e42ffe43c37161517f2ec05157d with all findings addressed; all review threads are resolved. Fast CI passed, and Full CI 35363166376 passed all 54 jobs on this head, including the previously failing Rust, wheel, Collector, and replay-contract checks. Current main remains d9f1580a and the reviewed branch includes it. Release publication and downstream Dynamo qualification remain separate gates.
